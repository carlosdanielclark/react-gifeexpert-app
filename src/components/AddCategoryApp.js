import React, { useState }  from 'react';
import '../stylesheet/AddCategoryApp.css';
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories, categories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleChange = (e) => {
    setInputValue(e.target.value);
  };
  
  //Manejo de tareas
  const handleSubmit = (e) => {
    e.preventDefault();
    
    //Validar dato
    if(inputValue.trim().length > 2){
      //Comunicacion de componentes
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <div className='addCategory_container'>
      <form className="task_form" onSubmit={handleSubmit}>
        <input
          type="text"
          name="category"
          id="category"
          placeholder="Add Title"
          value={inputValue}
          onChange={handleChange}
        />
        <button>add +</button>
      </form>
      <h1>{categories}</h1>
    </div>
  );
};

//Manejo de tipado en las propiedades
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};