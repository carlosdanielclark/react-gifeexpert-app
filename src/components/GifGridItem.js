import React from 'react';
import '../stylesheet/GifGridItem.css';
import "animate.css";
import PropTypes from 'prop-types'; // ES6

const GifGridItem = ({ title, url })=>{
  return (
    <div className="animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  );
}

export default GifGridItem;

GifGridItem.propTypes = {
  // You can declare that a prop is a specific JS primitive. By default, these
  // are all optional.
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,

}