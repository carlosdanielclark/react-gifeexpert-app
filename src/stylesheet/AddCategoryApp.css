.addCategory_container{
  display: flex;
  flex-direction: column;
}
.addCategory_container h1{
  text-align: center;
}
.form {
  margin-top: 2px;
}

input {
  color: antiquewhite;
  font-size: 20px;
  width: 80%;
  height: 29px;
  background-color: #21232A;
}

button {
  width: 65px;
  height: 30px;
  color: brown;
  text-transform: uppercase;
  font-style: italic;
  font-size: 18px;
  border-radius: 5px;
  background-color: azure;
}

