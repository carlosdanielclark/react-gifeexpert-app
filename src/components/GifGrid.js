import React from 'react';
import '../stylesheet/GifGrid.css'
import  GifGridItem from './GifGridItem.js';
import { useFetchGifs } from '../hooks/useFetchGif.js'; 
import { PropTypes } from 'prop-types';

const GifGrid = ({ category }) => {

  const { data:image, loading }=useFetchGifs( category );

  return (
    <> 
      <h3 className='animate_animated animate_fadeIn'>{ category }</h3>
      
      {loading && <p className="animate_animated animate_flash">Loading</p>}

      <div className='GifGrid_container'>
        {image.map((img) => {
          return <GifGridItem key={img.id} title={img.title} url={img.url} />;
        })}
      </div>
    </>
  );
}

export default GifGrid;

GifGrid.propTypes={
  category: PropTypes.string.isRequired
} 