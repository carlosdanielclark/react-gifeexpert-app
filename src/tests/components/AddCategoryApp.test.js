import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'; 
import { AddCategory } from '../../components/AddCategoryApp'

describe('Testing component "AddCategoryApp.js"', ()=>{
  //(1) Inicializacion
  const setCategories = jest.fn();
  const categories = "venom"; 

  //(2) Estimulo
  let wrapper 

  beforeEach(()=>{
    jest.clearAllMocks(); //Clean simulation
    wrapper = shallow( <AddCategory 
      setCategories ={ setCategories } 
      categories = { categories }
      /> );
  })
  
  test('Take Snapshot from "AddCategory"', () => {

    //(3) Observacion
    expect( wrapper ).toMatchSnapshot();

  });

  test('Must change the text box in the "input"', () => {

    const value = 'One puch'|| categories;

    //(3) Observacion
    // 1. simulate input-change
    wrapper.find('input').simulate('change', { target: { value } });
    // 2. set input-value is 'new_value'
   expect( wrapper.find('input').prop('value') ).toBe('One puch');

  });

  test('Should not post the information with "submit"', () => {
   
    //(3) Observacion
    // 1. simulate form-submit
    wrapper.find('form').simulate('submit', { preventDefault: ()=>{} });
    // 2. simulate setCategorie
    expect( setCategories ).not.toHaveBeenCalled();
    expect( setCategories ).toHaveBeenCalledTimes(0);
   
  });

  test('Must call "setCategories" and clean text box', () => {

    const value = 'Hunter X Hunter';

    //(3) Observacion
    // 1. simulate input-change
    wrapper.find('input').simulate('change', { target: { value } });
    // 2. simulate form-submit
    wrapper.find('form').simulate('submit', { preventDefault: ()=>{} });
    // 3. simulate setCategorie
    expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );
    // 4. set input-value in 'empty'
   expect( wrapper.find('input').prop('value') ).toBe('');
  });
  
});