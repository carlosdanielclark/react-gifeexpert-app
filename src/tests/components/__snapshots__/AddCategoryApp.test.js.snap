// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component "AddCategoryApp.js" Take Snapshot from "AddCategory" 1`] = `
<div
  className="addCategory_container"
>
  <form
    className="task_form"
    onSubmit={[Function]}
  >
    <input
      id="category"
      name="category"
      onChange={[Function]}
      placeholder="Add Title"
      type="text"
      value=""
    />
    <button>
      add +
    </button>
  </form>
</div>
`;
